airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py

```diff
@@ -104,102 +137,19 @@ def request_params(self, stream_state: Mapping[str, Any] = None, next_page_token
-    primary_key = "id"
-    page_size = 100
-
-    transformer = TypeTransformer(TransformConfig.CustomSchemaNormalization)
-
-    def __init__(self, api_key: str, **kwargs):
-        super().__init__(**kwargs)
-        self._api_key = api_key
-        transform_function = self.get_custom_transform()
-        self.transformer.registerCustomTransform(transform_function)
-
-    @staticmethod
-    def get_custom_transform():
-        def custom_transform_date_rfc3339(original_value, field_schema):
-            if original_value and "format" in field_schema and field_schema["format"] == "date-time":
-                transformed_value = pendulum.parse(original_value).to_rfc3339_string()
-                return transformed_value
-            return original_value
 
-        return custom_transform_date_rfc3339
-
-    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        """
-        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests.
-        This dict is passed to most other methods in this class to help you form headers, request bodies, query params, etc.
-        :param response: the most recent response from the API
-        :return If there is another page in the result, a mapping (e.g: dict) containing info needed to query the next page in the response.
-                If there are no more pages in the result, return None.
-        """
-        decoded_response = response.json()
-        if decoded_response["end"] < decoded_response["total"] - 1:  # end is zero based
-            return {"page": decoded_response["page"] + 1}
-
-        return None
-
-    def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
-        """Usually contains common params e.g. pagination size etc."""
-        next_page_token = next_page_token or {}
-        return {**next_page_token, "api_key": self._api_key, "count": self.page_size}
-
-    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        """:return an iterable containing each record in the response"""
-
-        response_json = response.json()
-        for record in response_json.get("data", []):  # API returns records in a container array "data"
-            yield record
-
-
-class IncrementalKlaviyoStreamV1(KlaviyoStreamV1, ABC):
-    """Base class for all incremental streams, requires cursor_field to be declared"""
-
-    def __init__(self, start_date: Optional[str], **kwargs):
-        super().__init__(**kwargs)
-        self._start_ts = int(pendulum.parse(start_date).timestamp()) if start_date else 0
-        self._start_sync = int(pendulum.now().timestamp())
+class SemiIncrementalKlaviyoStream(KlaviyoStream, ABC):
+    """Base class for all streams that natively incremental, but either do not have sorting or filtering"""
```

Comment:
I am having a hard time understanding this

Reply:
Sorry about that, I've updated the description string. Please check